Problem1.java

```diff
@@ -0,0 +1,62 @@
+//Time Complexity:O(logn)
+//Space Complexity:O(1)
+//Did it run on Leetcode: Yes
+//Any issues: The program compiled and executed successfully; All the visible test cases have passes; Although, for the hidden test cases, there was a time exceed alert
```

Comment:
here is a a cleaner solution 

class Solution {
    public int[] searchRange(int[] nums, int target) {
        int[] result  = new int[2];
        result[0] = findFirstIndex(nums, target);
        result[1] = findLastIndex(nums, target);
        return result;
    }
    private int findFirstIndex(int[] nums, int target)
    {
        int left = 0;
        int right = nums.length - 1;
        int index = -1;
        while(left <= right)
        {
            int mid = left + (right - left)/2;
            if (nums[mid] >= target )
            {
                right = mid -1;
            }
            else
            {
                left = mid + 1;
            }
            if(target == nums[mid])
            {
                index = mid;
            }

            

        
        }
        return index;
    }
    
    private int findLastIndex(int[] nums, int target)
    {
        int index = -1;
        int left = 0;
        int right = nums.length - 1;
        while(left <= right)
        {
            int mid = left + (right - left)/2;
           
            if (nums[mid] <= target )
            {
                left = mid + 1;
            }
            else
            {
                right = mid - 1;
            }
             if(target == nums[mid])
            {
                index = mid;
            }
        }
        return index;
    }
}